lib/ex_unit/test/ex_unit_test.exs

```diff
@@ -288,23 +288,41 @@ defmodule ExUnitTest do
     assert output =~ "2 tests, 1 failure, 1 skipped"
   end
 
-  test "raises on reserved tag in module" do
+  test "raises on reserved tag :file in module" do
     assert_raise RuntimeError, "cannot set tag :file because it is reserved by ExUnit", fn ->
-      defmodule ReservedTag do
+      defmodule ReservedTagFile do
         use ExUnit.Case
 
-        setup do
-          {:ok, file: :foo}
-        end
-
         @tag file: "oops"
         test "sample", do: :ok
       end
     end
   end
 
-  test "raises on reserved tag in setup" do
-    defmodule ReservedSetupTag do
+  test "raises on reserved tag :async in module" do
+    assert_raise RuntimeError, "cannot set tag :async because it is reserved by ExUnit", fn ->
+      defmodule ReservedTagAsync do
+        use ExUnit.Case
+
+        @tag async: true
+        test "sample", do: :ok
+      end
+    end
+  end
+
+  test "raises on reserved tag :registered in module" do
```

Comment:
Added a few tests here to make sure the changes were effective. However, this might be considered a little overtesting, since they are pretty similar in their essence. Other reserved tags, namely `case`, `line` and `test` are still uncovered, though.